data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.amazon.dataprepper.pipeline.server;
+
+import io.micrometer.cloudwatch2.CloudWatchConfig;
+import io.micrometer.cloudwatch2.CloudWatchMeterRegistry;
+import io.micrometer.core.instrument.Clock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * Provides {@link CloudWatchMeterRegistry} that enables publishing metrics to AWS Cloudwatch. Registry
+ * uses the default aws credentials (i.e. credentials from .aws directory;
+ * refer https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/credentials.html#credentials-file-format).
+ * {@link CloudWatchMeterRegistryProvider} also has a constructor with {@link CloudWatchAsyncClient} that will be used
+ * for communication with Cloudwatch.
+ */
+public class CloudWatchMeterRegistryProvider {
+    private static final String CLOUDWATCH_PROPERTIES = "cloudwatch.properties";
+    private static final Logger LOG = LoggerFactory.getLogger(CloudWatchMeterRegistryProvider.class);
+
+    private final CloudWatchMeterRegistry cloudWatchMeterRegistry;
+
+    public CloudWatchMeterRegistryProvider() {
+        this(CLOUDWATCH_PROPERTIES, CloudWatchAsyncClient.create());
+    }
+
+    public CloudWatchMeterRegistryProvider(
+            final String cwPropertiesFile,
+            final CloudWatchAsyncClient cloudWatchClient) {
+        final CloudWatchConfig cloudWatchConfig = createCloudWatchConfig(cwPropertiesFile);
+        this.cloudWatchMeterRegistry = new CloudWatchMeterRegistry(cloudWatchConfig, Clock.SYSTEM, cloudWatchClient);
+    }
+
+    /**
+     * Returns the CloudWatchMeterRegistry created using the default aws credentials
+     */
+    public CloudWatchMeterRegistry getCloudWatchMeterRegistry() {
+        return this.cloudWatchMeterRegistry;
+    }
+
+    /**
+     * Returns CloudWatchConfig using the properties from {@link #CLOUDWATCH_PROPERTIES}
+     */
+    private CloudWatchConfig createCloudWatchConfig(final String cwPropertiesFile) {
+        CloudWatchConfig cloudWatchConfig = null;
+        try (final InputStream inputStream = Objects.requireNonNull(getClass()
+                .getClassLoader().getResourceAsStream(cwPropertiesFile))) {
+            final Properties cloudwatchProperties = new Properties();
+            cloudwatchProperties.load(inputStream);
+            cloudWatchConfig = cloudwatchProperties::getProperty; //overriding getKey() from CloudWatchConfig
```

Comment:
Just curious - I haven't seen `::` used like this before, what's going on here?

Reply:
overriding getKey() from CloudWatchConfig [Added comment as I expected this :-P ]

```
new CloudWatchConfig() {
                @Override
                public String get(String key) {
                    return cloudwatchProperties.getProperty(key);
                }
            }
```